src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionCodeFixProvider.cs

```diff
@@ -159,60 +158,46 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                 constraintClauses, body, expressionBody, semicolonToken);
         }
 
-        private ParameterListSyntax GenerateParameterList(
-            SemanticModel semanticModel, AnonymousFunctionExpressionSyntax anonymousFunction, INamedTypeSymbol delegateType, CancellationToken cancellationToken)
+        private static ParameterListSyntax GenerateParameterList(
+            AnonymousFunctionExpressionSyntax anonymousFunction, IMethodSymbol delegateMethod)
         {
-            switch (anonymousFunction)
-            {
-                case SimpleLambdaExpressionSyntax simpleLambda:
-                    return GenerateSimpleLambdaParameterList(semanticModel, simpleLambda, delegateType.DelegateInvokeMethod, cancellationToken);
-                case ParenthesizedLambdaExpressionSyntax parenthesizedLambda:
-                    return GenerateParenthesizedLambdaParameterList(semanticModel, parenthesizedLambda, delegateType.DelegateInvokeMethod, cancellationToken);
-                default:
-                    throw ExceptionUtilities.UnexpectedValue(anonymousFunction);
-            }
-        }
+            var parameterList = GetOrMakeParameterList(anonymousFunction);
+            int i = 0;
 
-        private ParameterListSyntax GenerateSimpleLambdaParameterList(
-            SemanticModel semanticModel, SimpleLambdaExpressionSyntax lambdaExpression, IMethodSymbol delegateInvokeMethod, CancellationToken cancellationToken)
-        {
-            var parameter = semanticModel.GetDeclaredSymbol(lambdaExpression.Parameter, cancellationToken);
-            var type = parameter?.Type.GenerateTypeSyntax() ?? s_objectType;
+            return parameterList != null
+                ? parameterList.ReplaceNodes(parameterList.Parameters, (parameterNode, _) => PromoteParameter(parameterNode, delegateMethod.Parameters[i++]))
+                : SyntaxFactory.ParameterList(SyntaxFactory.SeparatedList(delegateMethod.Parameters.Select(parameter =>
+                    PromoteParameter(SyntaxFactory.Parameter(parameter.Name.ToIdentifierToken()), parameter))));
 
-            var parameterSyntax = SyntaxFactory.Parameter(lambdaExpression.Parameter.Identifier).WithType(type);
-            var param = delegateInvokeMethod.Parameters[0];
-            if (param.HasExplicitDefaultValue)
+            ParameterSyntax PromoteParameter(ParameterSyntax parameterNode, IParameterSymbol delegateParameter)
```

Comment:
I noticed I'm not capturing any variables anymore, should it still be a local function?

Reply:
My thoughts are:  if the code is readable and maintainable (and you're not violating any guidelines), then code however you like.  WRT local functions, we've cautiously accepted them.  We don't go overboard with them, but we allow devs to use them if they feel like it's worthwhile.

--

My personal preference is that all local functions go *after* teh last return point in the method.  i.e. after the final ```return``` statement in the method.  i usually add a ```// local functions``` comment as well, as i find that helps distinguish these from normal methods.  Otherwise, when reviewing i have a tendency to then say "you're missing accessibility modifiers here" since it's harder to tell in github when something is a method vs local function.